experiments/conductor/cmd/runner/crd_generator_commands.go

```diff
@@ -0,0 +1,320 @@
+	workDir := opts.branchRepoDir
+
+	var out strings.Builder
+	checkoutBranch(branch, workDir, &out)
+
+	// Change to controllerbuilder directory
+	controllerBuilderDir := filepath.Join(workDir, "dev", "tools", "controllerbuilder")
+
+	// Generate types
+	cfg := CommandConfig{
+		Name: "Generate types",
+		Cmd:  "go",
+		Args: []string{
+			"run", ".",
+			"generate-types",
+			"--service", branch.Package,
+			"--api-version", fmt.Sprintf("%s.cnrm.cloud.google.com/v1alpha1", branch.Group),
+			"--resource", fmt.Sprintf("%s:%s", branch.Kind, branch.Proto),
+		},
+		WorkDir: controllerBuilderDir,
+	}
+	if err := executeCommand(cfg, &out); err != nil {
+		log.Fatal(err)
+	}
+
+	// Generate mapper
+	cfg = CommandConfig{
+		Name: "Generate mapper",
+		Cmd:  "go",
+		Args: []string{
+			"run", ".",
+			"generate-mapper",
+			"--service", branch.Package,
+			"--api-version", fmt.Sprintf("%s.cnrm.cloud.google.com/v1alpha1", branch.Group),
+		},
+		WorkDir: controllerBuilderDir,
+	}
+	if err := executeCommand(cfg, &out); err != nil {
+		log.Fatal(err)
+	}
+
+	// Run goimports
+	cfg = CommandConfig{
+		Name: "Format generated code",
+		Cmd:  "go",
+		Args: []string{
+			"run", "-mod=readonly",
+			"golang.org/x/tools/cmd/goimports@latest",
+			"-w",
+			fmt.Sprintf("pkg/controller/direct/%s/", branch.Group),
+		},
+		WorkDir: workDir,
+	}
+	if err := executeCommand(cfg, &out); err != nil {
+		log.Fatal(err)
+	}
+
+	// Add and commit generated files
+	gitAdd(workDir, &out,
+		fmt.Sprintf("apis/%s/v1alpha1/", branch.Group),
+		fmt.Sprintf("pkg/controller/direct/%s/", branch.Group))
+	gitCommit(workDir, &out, fmt.Sprintf("Generated types and mapper for %s", branch.Kind))
+}
+
+func generateCRD(opts *RunnerOptions, branch Branch) {
+	close := setLoggingWriter(opts, branch)
+	defer close()
+	workDir := filepath.Join(opts.branchRepoDir)
+
+	var out strings.Builder
+	checkoutBranch(branch, workDir, &out)
+
+	// Generate CRDs
+	cfg := CommandConfig{
+		Name:    "Generate CRDs",
+		Cmd:     filepath.Join(workDir, "dev/tasks/generate-crds"),
+		WorkDir: workDir,
+	}
+	if err := executeCommand(cfg, &out); err != nil {
+		log.Fatal(err)
+	}
+
+	// Add and commit generated files
+	gitAdd(workDir, &out, "config/crds/resources/")
+	gitCommit(workDir, &out, fmt.Sprintf("Generated CRD for %s", branch.Kind))
+}
+
+func generateSpecStatus(opts *RunnerOptions, branch Branch) {
+	close := setLoggingWriter(opts, branch)
+	defer close()
+	workDir := filepath.Join(opts.branchRepoDir)
+
+	var out strings.Builder
+	checkoutBranch(branch, workDir, &out)
+
+	// Run controllerbuilder to generate spec and status
+	log.Printf("Generating spec and status for %s", branch.Name)
+	stdinInput := fmt.Sprintf("// +kcc:proto=%s.%s\n", branch.ProtoSvc, branch.Proto)
+
+	cfg := CommandConfig{
+		Name:    "Spec/Status generation",
+		Cmd:     "controllerbuilder",
+		Args:    []string{"prompt", "--src-dir", workDir, "--proto-dir", filepath.Join(workDir, ".build/third_party/googleapis/")},
```

Comment:
Do you have an example commit generated by this command? My experience is that this step currently does not work?